### PR TITLE
Enable HMR depending on debug mode

### PR DESCRIPTION
**Pull request by onny**
## Summary

Currently Nextlouc needs to get patched to use Vue Devtools due to "hot module reloading" is restricted, see
- https://github.com/nextcloud-libraries/nextcloud-vue?tab=readme-ov-file#using-vue-devtools-in-firefox
- https://github.com/nextcloud/hmr_enabler

This PR loosens security on debug mode and makes Vue Devtools usable

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by onny** [comment]
ready for review 😊 @skjnldsv @artonge

File: lib/private/AppFramework/Middleware/Security/CSPMiddleware.php [hunk 1]
```diff
@@ -74,6 +74,17 @@ public function afterController($controller, $methodName, Response $response): R
 			$defaultPolicy->useJsNonce($this->csrfTokenManager->getToken()->getEncryptedValue());
 		}
 
+		// Loosen security presets in debug mode to enable development
+		// tools functionality
+		$debugging = \OC::$server->getConfig()->getSystemValue('debug', false);
```
Comment (skjnldsv) [inline lib/private/AppFramework/Middleware/Security/CSPMiddleware.php:79]: please use dependency injection in the construct :)
`\OC::$server->getConfig()` is deprecated

**Review by ChristophWurst — CHANGES_REQUESTED** [review]
I am strongly against this. Most developers have their dev setups in debug mode and never test without. So we'll likely run into features breaking in production because nobody notices CSP errors locally.

**Comment by onny** [comment]
> I am strongly against this. Most developers have their dev setups in debug mode and never test without. So we'll likely run into features breaking in production because nobody notices CSP errors locally.

would it be possible to introduce an extra system setting variable like "enable_hmr" which can be only enabled if debug mode is also enabled?

**Comment by ShGKme** [comment]
HMR requires connection to only one server on a specific host. A proper solution would be to fix only required restrictions for a specific dev server and not to disable CSP completely.

**Comment by ShGKme** [comment]
I'd personally prefer to have a solution in a separated app, not in the server.

**Pull request closed without merging**